File: 500/598 - Range Addition II.py
class Solution:
    def maxCount(self, m, n, ops):
        """
        :type m: int
        :type n: int
        :type ops: List[List[int]]
        :rtype: int
        """
        if not ops:
            return m * n
        return min(m, min([x[0] for x in ops])) * min(n, min([x[1] for x in ops]));

File: csharp/598.range-addition-ii.cs
/*
 * @lc app=leetcode id=598 lang=csharp
 *
 * [598] Range Addition II
 */

using System.Linq;

// @lc code=start
public class Solution {
    public int MaxCount(int m, int n, int[][] ops) {
        return ops.Aggregate((m, n), (pair, curr) => (Math.Min(pair.Item1, curr[0]), Math.Min(pair.Item2, curr[1])), pair => pair.Item1 * pair.Item2);
    }
}
// @lc code=end

